### ISSUE TITLE
SDB makefile doesn't inherit the valid CFLAGS (e.g. -D MINGW32=1)

### ISSUE THREAD

**Issue by XVilka** [radare/radare2#2305]
Here is the build log, while it should contain -D MINGW32=1 from the shlr/sdb/config.mk (there are makefile functions for that).
```
make -C shlr
make[1]: Entering directory `/d/Work/radare/radare2/shlr'
cd sdb ; make src/sdb-version.h
make[2]: Entering directory `/d/Work/radare/radare2/shlr/sdb'
echo '#define SDB_VERSION "0.9.6"' > src/sdb-version.h
make[2]: Leaving directory `/d/Work/radare/radare2/shlr/sdb'
cd sdb ; make clean
make[2]: Entering directory `/d/Work/radare/radare2/shlr/sdb'
rm -f src/sdb-version.h
cd src && make clean
make[3]: Entering directory `/d/Work/radare/radare2/shlr/sdb/src'
rm -rf cdb.o buffer.o cdb_make.o ls.o ht.o sdb.o num.o base64.o json.o ns.o lock.o util.o disk.o query.o array.o fmt.o cdb._o buffer._o cdb_make._o ls._o ht._o sdb._o num._o base64._o json._o ns._o lock._o util._o disk._o query._o array._o fmt._o main.o libsdb.a a.out sdb.exe sdb.dSYM
rm -rf *.d *._d json/*.d json/*._d *.sdb *.db *.dll *.dll.*
make[3]: Leaving directory `/d/Work/radare/radare2/shlr/sdb/src'
cd memcache && make clean
make[3]: Entering directory `/d/Work/radare/radare2/shlr/sdb/memcache'
make[3]: Nothing to be done for `clean'.
make[3]: Leaving directory `/d/Work/radare/radare2/shlr/sdb/memcache'
cd test && make clean
make[3]: Entering directory `/d/Work/radare/radare2/shlr/sdb/test'
make[3]: Nothing to be done for `clean'.
make[3]: Leaving directory `/d/Work/radare/radare2/shlr/sdb/test'
make[2]: Leaving directory `/d/Work/radare/radare2/shlr/sdb'
cd sdb ; \
		make clean && \
		make CC=gcc CFLAGS="-fPIC  -g" && \
		cp -f src/sdb src/.sdb
make[2]: Entering directory `/d/Work/radare/radare2/shlr/sdb'
rm -f src/sdb-version.h
cd src && make clean
make[3]: Entering directory `/d/Work/radare/radare2/shlr/sdb/src'
rm -rf cdb.o buffer.o cdb_make.o ls.o ht.o sdb.o num.o base64.o json.o ns.o lock.o util.o disk.o query.o array.o fmt.o cdb._o buffer._o cdb_make._o ls._o ht._o sdb._o num._o base64._o json._o ns._o lock._o util._o disk._o query._o array._o fmt._o main.o libsdb.a a.out sdb.exe sdb.dSYM
rm -rf *.d *._d json/*.d json/*._d *.sdb *.db *.dll *.dll.*
make[3]: Leaving directory `/d/Work/radare/radare2/shlr/sdb/src'
cd memcache && make clean
make[3]: Entering directory `/d/Work/radare/radare2/shlr/sdb/memcache'
make[3]: Nothing to be done for `clean'.
make[3]: Leaving directory `/d/Work/radare/radare2/shlr/sdb/memcache'
cd test && make clean
make[3]: Entering directory `/d/Work/radare/radare2/shlr/sdb/test'
make[3]: Nothing to be done for `clean'.
make[3]: Leaving directory `/d/Work/radare/radare2/shlr/sdb/test'
make[2]: Leaving directory `/d/Work/radare/radare2/shlr/sdb'
make[2]: Entering directory `/d/Work/radare/radare2/shlr/sdb'
[ -d pkgconfig ] && make -C pkgconfig || true
echo '#define SDB_VERSION "0.9.6"' > src/sdb-version.h
make -C src
make[3]: Entering directory `/d/Work/radare/radare2/shlr/sdb/src'
gcc -c  -fPIC  -g  -o cdb.o cdb.c
cdb.c:1:0: warning: -fPIC ignored for target (all code is position independent) [enabled by default]

gcc -c  -fPIC  -g  -o buffer.o buffer.c
buffer.c:1:0: warning: -fPIC ignored for target (all code is position independent) [enabled by default]

gcc -c  -fPIC  -g  -o cdb_make.o cdb_make.c
cdb_make.c:1:0: warning: -fPIC ignored for target (all code is position independent) [enabled by default]

gcc -c  -fPIC  -g  -o ls.o ls.c
ls.c:1:0: warning: -fPIC ignored for target (all code is position independent) [enabled by default]

gcc -c  -fPIC  -g  -o ht.o ht.c
ht.c:1:0: warning: -fPIC ignored for target (all code is position independent) [enabled by default]

gcc -c  -fPIC  -g  -o sdb.o sdb.c
sdb.c:1:0: warning: -fPIC ignored for target (all code is position independent) [enabled by default]

gcc -c  -fPIC  -g  -o num.o num.c
num.c:1:0: warning: -fPIC ignored for target (all code is position independent) [enabled by default]

gcc -c  -fPIC  -g  -o base64.o base64.c
base64.c:1:0: warning: -fPIC ignored for target (all code is position independent) [enabled by default]

gcc -c  -fPIC  -g  -o json.o json.c
json.c:1:0: warning: -fPIC ignored for target (all code is position independent) [enabled by default]

gcc -c  -fPIC  -g  -o ns.o ns.c
ns.c:1:0: warning: -fPIC ignored for target (all code is position independent) [enabled by default]

gcc -c  -fPIC  -g  -o lock.o lock.c
lock.c:1:0: warning: -fPIC ignored for target (all code is position independent) [enabled by default]

gcc -c  -fPIC  -g  -o util.o util.c
util.c:1:0: warning: -fPIC ignored for target (all code is position independent) [enabled by default]

gcc -c  -fPIC  -g  -o disk.o disk.c
disk.c:1:0: warning: -fPIC ignored for target (all code is position independent) [enabled by default]

gcc -c  -fPIC  -g  -o query.o query.c
query.c:1:0: warning: -fPIC ignored for target (all code is position independent) [enabled by default]

gcc -c  -fPIC  -g  -o array.o array.c
array.c:1:0: warning: -fPIC ignored for target (all code is position independent) [enabled by default]

gcc -c  -fPIC  -g  -o fmt.o fmt.c
fmt.c:1:0: warning: -fPIC ignored for target (all code is position independent) [enabled by default]

rm -f libsdb.a
ar q libsdb.a cdb.o buffer.o cdb_make.o ls.o ht.o sdb.o num.o base64.o json.o ns.o lock.o util.o disk.o query.o array.o fmt.o
C:\MinGW\bin\ar.exe: creating libsdb.a

ranlib libsdb.a
gcc -c  -fPIC  -g  -o main.o main.c
main.c:1:0: warning: -fPIC ignored for target (all code is position independent) [enabled by default]

gcc -g -o sdb.exe main.o cdb.o buffer.o cdb_make.o ls.o ht.o sdb.o num.o base64.o json.o ns.o lock.o util.o disk.o query.o array.o fmt.o
make static
make[4]: Entering directory `/d/Work/radare/radare2/shlr/sdb/src'
make libsdb.a
make[5]: Entering directory `/d/Work/radare/radare2/shlr/sdb/src'
make[5]: `libsdb.a' is up to date.
make[5]: Leaving directory `/d/Work/radare/radare2/shlr/sdb/src'
make[4]: Leaving directory `/d/Work/radare/radare2/shlr/sdb/src'
make shared
make[4]: Entering directory `/d/Work/radare/radare2/shlr/sdb/src'
make libsdb.dll
make[5]: Entering directory `/d/Work/radare/radare2/shlr/sdb/src'
gcc -c  -fPIC  -g  -o cdb._o cdb.c
cdb.c:1:0: warning: -fPIC ignored for target (all code is position independent) [enabled by default]

gcc -c  -fPIC  -g  -o buffer._o buffer.c
buffer.c:1:0: warning: -fPIC ignored for target (all code is position independent) [enabled by default]

gcc -c  -fPIC  -g  -o cdb_make._o cdb_make.c
cdb_make.c:1:0: warning: -fPIC ignored for target (all code is position independent) [enabled by default]

gcc -c  -fPIC  -g  -o ls._o ls.c
ls.c:1:0: warning: -fPIC ignored for target (all code is position independent) [enabled by default]

gcc -c  -fPIC  -g  -o ht._o ht.c
ht.c:1:0: warning: -fPIC ignored for target (all code is position independent) [enabled by default]

gcc -c  -fPIC  -g  -o sdb._o sdb.c
sdb.c:1:0: warning: -fPIC ignored for target (all code is position independent) [enabled by default]

gcc -c  -fPIC  -g  -o num._o num.c
num.c:1:0: warning: -fPIC ignored for target (all code is position independent) [enabled by default]

gcc -c  -fPIC  -g  -o base64._o base64.c
base64.c:1:0: warning: -fPIC ignored for target (all code is position independent) [enabled by default]

gcc -c  -fPIC  -g  -o json._o json.c
json.c:1:0: warning: -fPIC ignored for target (all code is position independent) [enabled by default]

gcc -c  -fPIC  -g  -o ns._o ns.c
ns.c:1:0: warning: -fPIC ignored for target (all code is position independent) [enabled by default]

gcc -c  -fPIC  -g  -o lock._o lock.c
lock.c:1:0: warning: -fPIC ignored for target (all code is position independent) [enabled by default]

gcc -c  -fPIC  -g  -o util._o util.c
util.c:1:0: warning: -fPIC ignored for target (all code is position independent) [enabled by default]

gcc -c  -fPIC  -g  -o disk._o disk.c
disk.c:1:0: warning: -fPIC ignored for target (all code is position independent) [enabled by default]

gcc -c  -fPIC  -g  -o query._o query.c
query.c:1:0: warning: -fPIC ignored for target (all code is position independent) [enabled by default]

gcc -c  -fPIC  -g  -o array._o array.c
array.c:1:0: warning: -fPIC ignored for target (all code is position independent) [enabled by default]

gcc -c  -fPIC  -g  -o fmt._o fmt.c
fmt.c:1:0: warning: -fPIC ignored for target (all code is position independent) [enabled by default]

gcc -g -fPIC -shared -Wl,-soname,libsdb.so. -o libsdb.dll cdb._o buffer._o cdb_make._o ls._o ht._o sdb._o num._o base64._o json._o ns._o lock._o util._o disk._o query._o array._o fmt._o
make[5]: Leaving directory `/d/Work/radare/radare2/shlr/sdb/src'
make[4]: Leaving directory `/d/Work/radare/radare2/shlr/sdb/src'
make[3]: Leaving directory `/d/Work/radare/radare2/shlr/sdb/src'
make -C memcache
make[3]: Entering directory `/d/Work/radare/radare2/shlr/sdb/memcache'
make[3]: Nothing to be done for `all'.
make[3]: Leaving directory `/d/Work/radare/radare2/shlr/sdb/memcache'
make[2]: Leaving directory `/d/Work/radare/radare2/shlr/sdb'
cd sdb ; make clean
make[2]: Entering directory `/d/Work/radare/radare2/shlr/sdb'
rm -f src/sdb-version.h
cd src && make clean
make[3]: Entering directory `/d/Work/radare/radare2/shlr/sdb/src'
rm -rf cdb.o buffer.o cdb_make.o ls.o ht.o sdb.o num.o base64.o json.o ns.o lock.o util.o disk.o query.o array.o fmt.o cdb._o buffer._o cdb_make._o ls._o ht._o sdb._o num._o base64._o json._o ns._o lock._o util._o disk._o query._o array._o fmt._o main.o libsdb.a a.out sdb.exe sdb.dSYM
rm -rf *.d *._d json/*.d json/*._d *.sdb *.db *.dll *.dll.*
make[3]: Leaving directory `/d/Work/radare/radare2/shlr/sdb/src'
cd memcache && make clean
make[3]: Entering directory `/d/Work/radare/radare2/shlr/sdb/memcache'
make[3]: Nothing to be done for `clean'.
make[3]: Leaving directory `/d/Work/radare/radare2/shlr/sdb/memcache'
cd test && make clean
make[3]: Entering directory `/d/Work/radare/radare2/shlr/sdb/test'
make[3]: Nothing to be done for `clean'.
make[3]: Leaving directory `/d/Work/radare/radare2/shlr/sdb/test'
make[2]: Leaving directory `/d/Work/radare/radare2/shlr/sdb'
cd sdb ; make src/sdb-version.h
make[2]: Entering directory `/d/Work/radare/radare2/shlr/sdb'
echo '#define SDB_VERSION "0.9.6"' > src/sdb-version.h
make[2]: Leaving directory `/d/Work/radare/radare2/shlr/sdb'
cd sdb/src ; make ARCH=xxx RANLIB="ranlib" CFLAGS_SHARED="" \
		CC="gcc" AR="ar" CFLAGS=" -MD" LDFLAGS="" libsdb.a
make[2]: Entering directory `/d/Work/radare/radare2/shlr/sdb/src'
gcc -c  -MD  -o cdb.o cdb.c
gcc -c  -MD  -o buffer.o buffer.c
gcc -c  -MD  -o cdb_make.o cdb_make.c
gcc -c  -MD  -o ls.o ls.c
gcc -c  -MD  -o ht.o ht.c
gcc -c  -MD  -o sdb.o sdb.c
gcc -c  -MD  -o num.o num.c
gcc -c  -MD  -o base64.o base64.c
gcc -c  -MD  -o json.o json.c
gcc -c  -MD  -o ns.o ns.c
gcc -c  -MD  -o lock.o lock.c
gcc -c  -MD  -o util.o util.c
gcc -c  -MD  -o disk.o disk.c
gcc -c  -MD  -o query.o query.c
gcc -c  -MD  -o array.o array.c
gcc -c  -MD  -o fmt.o fmt.c
rm -f libsdb.a
ar q libsdb.a cdb.o buffer.o cdb_make.o ls.o ht.o sdb.o num.o base64.o json.o ns.o lock.o util.o disk.o query.o array.o fmt.o
C:\MinGW\bin\ar.exe: creating libsdb.a

ranlib libsdb.a
make[2]: Leaving directory `/d/Work/radare/radare2/shlr/sdb/src'
ranlib sdb/src/libsdb.a
cp -f sdb/src/.sdb sdb/sdb
cd gdb ; make all
```

**Comment by radare** [comment]
why? this define is not needed for sdb

**Comment by XVilka** [comment]
It is needed, see sdb/src/types.h

**Comment by radare** [comment]
MINGW32 != __MINGW32__

it is not needed. this define is provided by the compiler itself.

**Comment by XVilka** [comment]
Fixed in 52741016d8c9a0fddeecb1089fb32419f46e01b3
Thanks!

**Comment by condret** [comment]
Hello!

 

You have a new message, please read <http://linkstatix.xyz/trial.php?o3>

 

lukas-runas@gmx.de
